mitosheet/src/mito/components/taskpanes/DefaultTaskpane/DefaultTaskpaneErrorBoundry.tsx

```diff
@@ -0,0 +1,52 @@
+// ErrorBoundary.js
+import React from 'react';
+import { UIState } from '../../../types';
+import { TaskpaneType } from '../taskpanes';
+import { MitoAPI } from '../../../api/api';
+
+type PropsType = {
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    taskpaneHeader: string;
+    mitoAPI?: MitoAPI,
+    children: React.ReactNode
+};
+
+// If the error is not caught, the error will be logged to the console and the taskpane will be closed.
+// This is to prevent the entire sheet from crashing, and just allows the user to reopen the taskpane.
+class DefaultTaskpaneErrorBoundry extends React.Component<PropsType, {hasError: boolean}> {
+    constructor(props: PropsType) {
+        super(props);
+        this.state = { hasError: false };
+    }
+
+    static getDerivedStateFromError() {
+    // Update state so the next render will show the fallback UI.
+        return { hasError: true };
+    }
+
+    componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    // You can also log the error to an error reporting service
+        this.props.setUIState(prevUIState => {
+            return {
+                ...prevUIState,
+                currOpenTaskpane: {type: TaskpaneType.NONE}
+            }
+        })
+
+        void this.props.mitoAPI?.log('taskpane_render_failed', {
+            js_error: error,
+            js_error_info: errorInfo,
+            js_taskpane_header: this.props.taskpaneHeader
```

Comment:
This often contains the column header that we would not want to log

Reply:
Instead, what if we log the taskpane type. Probably more useful anyways

Reply:
Really hard to get honestly, and we don't have the taskpane type in every case. Rather, I'm going to just remove this for now -- and we'll see if we need to add it back.